core/includes/common.inc

```diff
@@ -8743,12 +8743,18 @@ function backdrop_check_incompatibility(array $dependency_info, $current_version
  * Converts a Backdrop version string into numeric-only version string.
  *
  * @param string $version_string
- *   A version string such as 1.10.0-beta4 or 1.4.x-dev.
+ *   A version string such as 1.x-1.2.3, or 1.10.0-beta4, or 1.4.x-dev.
  * @return string
  *   A converted string only containing numbers, for use in PHP's
  *   version_compare() function.
  */
 function _backdrop_version_compare_convert($version_string) {
+  // Remove the "1.x-" prefix (indicating Backdrop core version compatibility).
+  $core_prefix = BACKDROP_CORE_COMPATIBILITY . '-';
+  if (substr($version_string, 0, strlen($core_prefix)) == $core_prefix) {
```

Comment:
This seems simpler:

```suggestion
  if (strpos($version_string, $core_prefix) === 0) {
```

Reply:
Indeed 😅 

Committed with [`a896e1e` (#3878)](https://github.com/backdrop/backdrop/pull/3878/commits/a896e1ea5e97e37cb5b29a44ba485d69e6e8bbf2)